corehq/apps/user_importer/importer.py

```diff
@@ -790,7 +790,7 @@ def remove_web_user_from_domain(domain, user, username, upload_user, user_change
                 user_change_logger.add_info(UserChangeMessage.invitation_revoked_for_domain(domain))
         else:
             raise UserUploadError(_("You cannot remove a web user that is not a member of this project."))
-    elif username == upload_user.username:
+    elif user.username == upload_user.username:
```

Comment:
This was working fine for a web upload where username and user's username were the same but in case of a mobile users upload where the user is the corresponding web user but the username is the username of the mobile worker, this check was not correct since this is `remove_web_user_from_domain`. So this now catches if the user doing the upload is passed as the web user for a mobile worker to be removed.
There was a test for the web upload, I added one for mobile upload which verified this change

Reply:
@stephherbers confirming that this change looks right?

Reply:
good catch-- yes this looks good and maintains the original intention